src/api/post/mutation.ts

```diff
@@ -56,19 +56,26 @@ export const usePostLikeMutation = (queryId: string) => {
   const queryClient = useQueryClient();
 
   return useMutation({
-    mutationKey: PostQueryKey.mutate(queryId),
     mutationFn: () => postPostLike(+queryId),
     onMutate: async () => {
-      const previousPost = queryClient.getQueryData(PostQueryKey.mutate(queryId)) as GetPostDetailResponse;
+      await queryClient.cancelQueries({ queryKey: PostQueryKey.detail(+queryId) });
 
-      const newLikeCount = previousPost.isLiked ? previousPost.likeCount - 1 : previousPost.likeCount + 1;
+      const previousPost = queryClient.getQueryData(PostQueryKey.detail(+queryId)) as GetPostDetailResponse;
```

Comment:
```suggestion
      const previousPost = queryClient.getQueryData<GetPostDetailResponse>(PostQueryKey.detail(+queryId));
```

타입 단언보다 제네릭이나 타입가드를 사용하면 좋을 것 같아요!

Reply:
`getQueryData`에도 제네릭을 저렇게 사용이 가능했네요..! 하나 배워갑니다!